waku/v2/node/storage/message/waku_message_store.nim

```diff
@@ -1,425 +1,269 @@
-    numRecordsMatchingPred = 0
-
-  let outPagingInfo = PagingInfo(pageSize: messages.len.uint,
-                             cursor: lastIndex,
-                             direction: pagingInfo.direction)
-
-  let historyResponseError = if numRecordsVisitedTotal == 0: HistoryResponseError.INVALID_CURSOR # Index is not in DB (also if queried Index points to last entry)
-    else: HistoryResponseError.NONE
-
-  preparedPageQuery.dispose()
-
-  return ok((messages, outPagingInfo, historyResponseError)) # TODO: historyResponseError is not a "real error": treat as a real error
-
-
-
-
-method getPage*(db: WakuMessageStore,
-              pagingInfo: PagingInfo):
-             MessageStoreResult[(seq[WakuMessage], PagingInfo, HistoryResponseError)] =
-  ## Get a single page of history without filtering.
-  ## Adhere to the pagingInfo parameters
-  
-  proc predicate(i: IndexedWakuMessage): bool = true # no filtering
+    let res = s.deleteMessagesExceedingRetentionTime()
+    if res.isErr(): 
+      return err("delete messages exceeding the retention time failed: " & res.error())
 
-  return getPage(db, predicate, pagingInfo)
+    # Update oldest timestamp after deleting messages
+    s.oldestReceiverTimestamp = s.db.selectOldestReceiverTimestamp().expect("query for oldest timestamp should work")
+    # Update message count after deleting messages
+    s.numMessages = int(s.db.getMessageCount().expect("query for oldest timestamp should work"))
 
+  ok()
 
 
+method getAllMessages*(s: WakuMessageStore):  MessageStoreResult[seq[MessageStoreRow]] =
+  ## Retrieve all messages from the store.
+  s.db.selectAllMessages()
+
+
+method getMessagesByHistoryQuery*(
+  s: WakuMessageStore,
+  contentTopic = none(seq[ContentTopic]),
+  pubsubTopic = none(string),
+  cursor = none(Index),
+  startTime = none(Timestamp),
+  endTime = none(Timestamp),
+  maxPageSize = StoreMaxPageSize,
+  ascendingOrder = true
+): MessageStoreResult[MessageStorePage] =
+  let pageSizeLimit = if maxPageSize <= 0: StoreMaxPageSize
+                      else: min(maxPageSize, StoreMaxPageSize)
+
+  let rows = ?s.db.selectMessagesByHistoryQueryWithLimit(
+    contentTopic, 
+    pubsubTopic, 
+    cursor,
+    startTime,
+    endTime,
+    limit=pageSizeLimit,
+    ascending=ascendingOrder
+  )
+
+  if rows.len <= 0:
+    return ok((@[], none(PagingInfo)))
+
+  var messages = rows.mapIt(it[0])
+
+  # TODO: Return the message hash from the DB, to avoid recomputing the hash of the last message
+  # Compute last message index
+  let (message, receivedTimestamp, pubsubTopic) = rows[^1]
+  let lastIndex = Index.compute(message, receivedTimestamp, pubsubTopic)
+
+  let pagingInfo = PagingInfo(
+    pageSize: uint64(messages.len),
+    cursor: lastIndex,
+    direction: if ascendingOrder: PagingDirection.FORWARD
+               else: PagingDirection.BACKWARD
+  )
+
+  # The retrieved messages list should always be in chronological order
+  if not ascendingOrder:
+    messages.reverse()
```

Comment:
These lines, as per @jm-clius comment, should solve the message ordering issue detected during the "dog-fooding" session.

Reply:
Great! Should fix it.